gpkit/geometric_program.py

```diff
@@ -174,8 +185,63 @@ def solve(self, solver=None, verbosity=1, *args, **kwargs):
                                  " in the 'result' attribute\n"
                                  "(model.program.result)"
                                  " and its raw output in 'solver_out'.")
-        else:
-            return result
+        
+        self.check_solution(result)
+        if verbosity > 1:
+            print ("solution checking took %.2g%% of solve time" %
+                   ((time() - tic) / soltime * 100))
+        return result
+
+    def check_solution(self, sol, tol=1e-5):
+        """Run a series of checks to mathematically confirm sol solves this GP
+
+        Arguments
+        ---------
+        sol: dict
+            solution dict, same format as return type of solve()
+
+        Raises
+        ------
+        RuntimeWarning, if any problems are found
+        """
+        def _almost_equal(num1, num2):
+            "local almost equal test"
+            return num1 == num2 or abs((num1 - num2) / (num1 + num2)) < tol
```

Comment:
Does it make sens to have this be the same `tol` as is used for distance from anything other than 1? (I think it's only used for 1 below, just checking)